pkg/client/userd/daemon/fuseftp.go

```diff
@@ -16,13 +16,52 @@ import (
 	"github.com/datawire/dlib/dlog"
 	"github.com/datawire/dlib/dtime"
 	"github.com/datawire/go-fuseftp/rpc"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
 	"github.com/telepresenceio/telepresence/v2/pkg/filelocation"
 	"github.com/telepresenceio/telepresence/v2/pkg/proc"
 )
 
 //go:embed fuseftp.bits
 var fuseftpBits []byte
 
+func (s *Service) deferFuseFtpInit(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return nil
+	case <-s.startFuseCh:
+	}
+	if client.GetConfig(ctx).Intercept.UseFtp {
+		if err := runFuseFTPServer(ctx, s.fuseFtpCh); err != nil {
+			dlog.Errorf(ctx, "Unable to start FuseFTP server: %v", err)
+		}
+	} else {
+		close(s.fuseFtpCh)
+	}
+	return nil
+}
+
+func (s *Service) GetFuseFTPClient(ctx context.Context) rpc.FuseFTPClient {
+	// Close startFuseFtp unless it's already closed. This will kick
+	// the deferFuseFtpInit to either make the client available on
+	// the fuseFtpCh or close that channel
+	select {
+	case <-s.startFuseCh:
+	default:
+		close(s.startFuseCh)
+	}
+
+	select {
+	case <-ctx.Done():
+		return nil
+	case c, ok := <-s.fuseFtpCh:
+		if ok {
+			// Put the client back onto the queue for the next caller to read
+			s.fuseFtpCh <- c
+		}
```

Comment:
interesting use of chan as storage

Reply:
I'm quite happy about it actually. It's a storage that requires the first reader to wait, and then synchronizes all subsequent reads without using a mutex.